src/service/synchronize.toit

```diff
@@ -376,20 +373,19 @@ class SynchronizeJob extends TaskJob:
       with-timeout TIMEOUT-NETWORK-OPEN: network = net.open
       while true:
         transition-to_ STATE-CONNECTED-TO-NETWORK
-        done := connect-broker_ network
+        broker := determine-broker_ --network=network
+        done := connect-broker_ --broker=broker --network=network
         if check-in:
           with-timeout TIMEOUT-CHECK-IN: check-in.run network logger_
         if done: return true
-        if Task.current.is-canceled:
```

Comment:
This was an attempt to find out why we hang, but nothing indicates we get stuck while canceled.